ydb/mvp/oidc_proxy/oidc_protected_page_nebius.h

```diff
@@ -0,0 +1,149 @@
+        TString sessionToken = Base64Decode(cookies.Get(CreateNameSessionCookie(Settings.ClientId)));
+
+        if (sessionToken) {
+            ExchangeSessionToken(sessionToken, ctx);
+        } else {
+            RequestAuthorizationCode(ctx);
+        }
+    }
+
+    void HandleProxy(NHttp::TEvHttpProxy::TEvHttpIncomingResponse::TPtr event, const NActors::TActorContext& ctx) {
+        NHttp::THttpOutgoingResponsePtr httpResponse;
+        if (event->Get()->Response != nullptr) {
+            NHttp::THttpIncomingResponsePtr response = event->Get()->Response;
+            LOG_DEBUG_S(ctx, EService::MVP, TStringBuilder() << "Incoming response for protected resource: " << response->Status);
+            if ((response->Status == "400" || response->Status.empty()) && RequestedPageScheme.empty()) {
+                NHttp::THttpOutgoingRequestPtr request = response->GetRequest();
+                if (!request->Secure) {
+                    LOG_DEBUG_S(ctx, EService::MVP, "Try to send request to HTTPS port");
+                    NHttp::THeadersBuilder headers {request->Headers};
+                    ForwardUserRequest(headers.Get(AUTH_HEADER_NAME), ctx, true);
+                    return;
+                }
+            }
+            NHttp::THeadersBuilder headers = GetResponseHeaders(response);
+            TStringBuf contentType = headers.Get("Content-Type").NextTok(';');
+            if (contentType == "text/html") {
+                TString newBody = FixReferenceInHtml(response->Body, response->GetRequest()->Host);
+                httpResponse = Request->CreateResponse( response->Status, response->Message, headers, newBody);
+            } else {
+                httpResponse = Request->CreateResponse( response->Status, response->Message, headers, response->Body);
+            }
+        } else {
+            httpResponse = Request->CreateResponseNotFound(NOT_FOUND_HTML_PAGE, "text/html");
+        }
+        ctx.Send(Sender, new NHttp::TEvHttpProxy::TEvHttpOutgoingResponse(httpResponse));
+        Die(ctx);
+    }
+
+    void HandleExchange(NHttp::TEvHttpProxy::TEvHttpIncomingResponse::TPtr event, const NActors::TActorContext& ctx) {
+        if (!event->Get()->Response) {
+            LOG_DEBUG_S(ctx, EService::MVP, "Getting access token: Bad Request");
+            NHttp::THeadersBuilder ResponseHeaders;
+            ResponseHeaders.Set("Content-Type", "text/plain");
+            NHttp::THttpOutgoingResponsePtr httpResponse = Request->CreateResponse("400", "Bad Request", ResponseHeaders, event->Get()->Error);
+            ctx.Send(Sender, new NHttp::TEvHttpProxy::TEvHttpOutgoingResponse(httpResponse));
+            Die(ctx);
+        } else {
+            NHttp::THttpIncomingResponsePtr response = event->Get()->Response;
+            LOG_DEBUG_S(ctx, EService::MVP, TStringBuilder() << "Getting access token: " << response->Status);
+            if (response->Status.StartsWith("2")) {
```

Comment:
What statuses we want to handle here? Only 200 or some additional?

Reply:
I will change it. probably for nebius 200 has to work fine